src/formats/ff.c

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+// farbfeld format decoder
+
+#include "../loader.h"
+#include <netinet/in.h>
+#include <string.h>
+
+enum loader_status decode_ff(struct image* ctx, const uint8_t* data,
+                              size_t size)
+{
+    size_t width, height;
+
+    if (size < 16 ) {
+        return ldr_unsupported;
+    }
+
+    static uint8_t magic[8] = {'f', 'a', 'r', 'b', 'f', 'e', 'l', 'd'};
+
+    if (memcmp(data, magic, 8) != 0) return ldr_unsupported;
+
+    memcpy(&width, (const uint8_t*) data+8, 4);
+    width = htonl(width);
+    memcpy(&height, (const uint8_t*) data+12, 4);
+    height = htonl(height);
+
+    // 16 bytes of metadata, plus farbfeld is exclusively a 16 bit format
+    if (16+width*height*sizeof(argb_t)*2 > size) return ldr_fmterror;
```

Comment:
By the way, why `*2`?

Reply:
argb_t assumes is 8 bit per channel, farbfeld is 16.

Reply:
farbfeld *only* supports 16 bit BE, and is assumed sRGB straight alpha. Since the image is supposed to be sRGB, I can just cut off the less significant bits and still have an sRGB image, just a less precise one.